src/dask_awkward/lib/inspect.py

```diff
@@ -81,8 +80,9 @@ def report_necessary_buffers(
 
     name_to_necessary_buffers: dict[str, NecessaryBuffers | None] = {}
     for obj in collections:
-        dsk = obj if isinstance(obj, HighLevelGraph) else obj.dask
-        projection_data = o._prepare_buffer_projection(dsk)
+        dsk = obj.__dask_graph__()
+        keys = obj.__dask_keys__()
+        projection_data = o._prepare_buffer_projection(dsk, keys)
 
```

Comment:
@douglasdavis first uncertainty - I'm pretty sure that anything in `collections` has to be something that has a `__dask_graph__` attribute, so I'm assuming here that we can also ask for `__dask_keys__` too.

Reply:
Yes `__dask_keys__` is part of the [collection interface](https://docs.dask.org/en/stable/_modules/dask/typing.html#DaskCollection.__dask_keys__)